### ISSUE TITLE
Cannot build codelite from source since SHA 419b168

### ISSUE THREAD

**Issue by uffejakobsen** [eranif/codelite#2673]
since SHA 419b168 I'm unable build codelite from source 

Seems that something related to the new Rust lexer did not make it into the source ? or ?

See build errors below

Problem seen on archlinux

```
make[2]: Entering directory '/tmp/_prj_local/codelite/codelite-devel.git/_build.out'
[ 15%] Building CXX object Plugin/CMakeFiles/plugin.dir/EclipseRustThemeImporter.cpp.o
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp: In member function ‘virtual LexerConf::Ptr_t EclipseRustThemeImporter::Import(const wxFileName&)’:
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:16:78: error: ‘wxSTC_LEX_RUST’ was not declared in this scope; did you mean ‘wxSTC_LEX_RUBY’?
   16 |     LexerConf::Ptr_t lexer = InitializeImport(eclipseXmlFile, GetLangName(), wxSTC_LEX_RUST);
      |                                                                              ^~~~~~~~~~~~~~
      |                                                                              wxSTC_LEX_RUBY
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:17:24: error: ‘wxSTC_RUST_DEFAULT’ was not declared in this scope; did you mean ‘wxSTC_ST_DEFAULT’?
   17 |     AddProperty(lexer, wxSTC_RUST_DEFAULT, "Default", m_foreground.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~~~
      |                        wxSTC_ST_DEFAULT
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:26:24: error: ‘wxSTC_RUST_LEXERROR’ was not declared in this scope; did you mean ‘wxSTC_RB_ERROR’?
   26 |     AddProperty(lexer, wxSTC_RUST_LEXERROR, "Lex error", errorColour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~~~~
      |                        wxSTC_RB_ERROR
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:27:24: error: ‘wxSTC_RUST_COMMENTBLOCK’ was not declared in this scope; did you mean ‘wxSTC_AVS_COMMENTBLOCK’?
   27 |     AddProperty(lexer, wxSTC_RUST_COMMENTBLOCK, "Comment block", m_multiLineComment.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~~~~~~~~
      |                        wxSTC_AVS_COMMENTBLOCK
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:28:24: error: ‘wxSTC_RUST_COMMENTLINE’ was not declared in this scope; did you mean ‘wxSTC_PAS_COMMENTLINE’?
   28 |     AddProperty(lexer, wxSTC_RUST_COMMENTLINE, "Comment line", m_singleLineComment.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~~~~~~~
      |                        wxSTC_PAS_COMMENTLINE
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:29:24: error: ‘wxSTC_RUST_COMMENTBLOCKDOC’ was not declared in this scope; did you mean ‘wxSTC_AVS_COMMENTBLOCK’?
   29 |     AddProperty(lexer, wxSTC_RUST_COMMENTBLOCKDOC, "Comment block doc", m_javadoc.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~~
      |                        wxSTC_AVS_COMMENTBLOCK
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:30:24: error: ‘wxSTC_RUST_COMMENTLINEDOC’ was not declared in this scope; did you mean ‘wxSTC_FS_COMMENTLINEDOC’?
   30 |     AddProperty(lexer, wxSTC_RUST_COMMENTLINEDOC, "Comment line doc", m_javadoc.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~
      |                        wxSTC_FS_COMMENTLINEDOC
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:31:24: error: ‘wxSTC_RUST_WORD’ was not declared in this scope; did you mean ‘wxSTC_BAT_WORD’?
   31 |     AddProperty(lexer, wxSTC_RUST_WORD, "Word", m_keyword.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~
      |                        wxSTC_BAT_WORD
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:32:24: error: ‘wxSTC_RUST_WORD2’ was not declared in this scope; did you mean ‘wxSTC_LUA_WORD2’?
   32 |     AddProperty(lexer, wxSTC_RUST_WORD2, "Word2", m_foreground.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~
      |                        wxSTC_LUA_WORD2
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:33:24: error: ‘wxSTC_RUST_WORD3’ was not declared in this scope; did you mean ‘wxSTC_LUA_WORD3’?
   33 |     AddProperty(lexer, wxSTC_RUST_WORD3, "Word3", m_foreground.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~
      |                        wxSTC_LUA_WORD3
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:34:24: error: ‘wxSTC_RUST_WORD4’ was not declared in this scope; did you mean ‘wxSTC_LUA_WORD4’?
   34 |     AddProperty(lexer, wxSTC_RUST_WORD4, "Word4", m_foreground.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~
      |                        wxSTC_LUA_WORD4
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:35:24: error: ‘wxSTC_RUST_WORD5’ was not declared in this scope; did you mean ‘wxSTC_LUA_WORD5’?
   35 |     AddProperty(lexer, wxSTC_RUST_WORD5, "Word5", m_foreground.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~
      |                        wxSTC_LUA_WORD5
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:36:24: error: ‘wxSTC_RUST_WORD6’ was not declared in this scope; did you mean ‘wxSTC_LUA_WORD6’?
   36 |     AddProperty(lexer, wxSTC_RUST_WORD6, "Word6", m_foreground.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~
      |                        wxSTC_LUA_WORD6
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:37:24: error: ‘wxSTC_RUST_WORD7’ was not declared in this scope; did you mean ‘wxSTC_LUA_WORD7’?
   37 |     AddProperty(lexer, wxSTC_RUST_WORD7, "Word7", m_foreground.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~
      |                        wxSTC_LUA_WORD7
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:38:24: error: ‘wxSTC_RUST_STRING’ was not declared in this scope; did you mean ‘wxSTC_ST_STRING’?
   38 |     AddProperty(lexer, wxSTC_RUST_STRING, "String", m_string.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~~
      |                        wxSTC_ST_STRING
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:39:24: error: ‘wxSTC_RUST_STRINGR’ was not declared in this scope; did you mean ‘wxSTC_ST_STRING’?
   39 |     AddProperty(lexer, wxSTC_RUST_STRINGR, "String raw", m_string.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~~~
      |                        wxSTC_ST_STRING
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:40:24: error: ‘wxSTC_RUST_CHARACTER’ was not declared in this scope; did you mean ‘wxSTC_ST_CHARACTER’?
   40 |     AddProperty(lexer, wxSTC_RUST_CHARACTER, "Character", m_string.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~~~~~
      |                        wxSTC_ST_CHARACTER
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:41:24: error: ‘wxSTC_RUST_OPERATOR’ was not declared in this scope; did you mean ‘wxSTC_SN_OPERATOR’?
   41 |     AddProperty(lexer, wxSTC_RUST_OPERATOR, "Operator", m_oper.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~~~~
      |                        wxSTC_SN_OPERATOR
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:42:24: error: ‘wxSTC_RUST_IDENTIFIER’ was not declared in this scope; did you mean ‘wxSTC_PAS_IDENTIFIER’?
   42 |     AddProperty(lexer, wxSTC_RUST_IDENTIFIER, "Identifier", m_variable.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~~~~~~
      |                        wxSTC_PAS_IDENTIFIER
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:43:24: error: ‘wxSTC_RUST_LIFETIME’ was not declared in this scope
   43 |     AddProperty(lexer, wxSTC_RUST_LIFETIME, "Lifetime", m_foreground.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~~~~
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:44:24: error: ‘wxSTC_RUST_MACRO’ was not declared in this scope; did you mean ‘wxSTC_AU3_MACRO’?
   44 |     AddProperty(lexer, wxSTC_RUST_MACRO, "Macro", wxColour(m_klass.colour).GetAsString(wxC2S_HTML_SYNTAX),
      |                        ^~~~~~~~~~~~~~~~
      |                        wxSTC_AU3_MACRO
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:46:24: error: ‘wxSTC_RUST_BYTESTRING’ was not declared in this scope; did you mean ‘wxSTC_ST_STRING’?
   46 |     AddProperty(lexer, wxSTC_RUST_BYTESTRING, "Byte string", m_string.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~~~~~~
      |                        wxSTC_ST_STRING
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:47:24: error: ‘wxSTC_RUST_BYTESTRINGR’ was not declared in this scope; did you mean ‘wxSTC_ST_STRING’?
   47 |     AddProperty(lexer, wxSTC_RUST_BYTESTRINGR, "Byte string raw", m_string.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~~~~~~~
      |                        wxSTC_ST_STRING
/tmp/prj.codelite/codelite-devel.git/Plugin/EclipseRustThemeImporter.cpp:48:24: error: ‘wxSTC_RUST_BYTECHARACTER’ was not declared in this scope; did you mean ‘wxSTC_ST_CHARACTER’?
   48 |     AddProperty(lexer, wxSTC_RUST_BYTECHARACTER, "Byte character", m_string.colour, m_background.colour);
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~
      |                        wxSTC_ST_CHARACTER
make[2]: *** [Plugin/CMakeFiles/plugin.dir/build.make:654: Plugin/CMakeFiles/plugin.dir/EclipseRustThemeImporter.cpp.o] Error 1

```

**Comment by datalot-369** [comment]
Seems to be the exact same error message I got from my first-time compile (https://github.com/eranif/codelite/issues/2674).
I solved it installing the last version of the Wx library (and cleaning up the previous 3.0 one).

@dghart said (https://github.com/eranif/codelite/issues/2674#issuecomment-780140511):

> Ah, I see: you're building against wx3.0.4 and that hasn't heard of wxSTC_LEX_RUST etc; it's new in wx3.1.
>
> That needs fixing in CodeLite, of course. Thank you for reporting it. Meanwhile, if you wish to build from git HEAD you could install wx3.1.4 (perhaps from https://wiki.codelite.org/pmwiki.php/Main/WxWidgets31Binaries, depending on your distro) or build it yourself. Otherwise you could revert to CodeLite 15.0.

**Comment by uffejakobsen** [comment]
Ah yes that makes sense

Unfortunately wxgtk3-3.1.x have not yet made it into my distribution - so that leaves me with some work to do...

**Comment by dghart** [comment]
This should now be fixed by #a10276df1

**Comment by uffejakobsen** [comment]
> This should now be fixed by #a10276df1

@dghart thanks :-)
